symengine/lib/symengine_wrapper.pyx

```diff
@@ -1161,6 +1178,46 @@ cdef class Complex(Number):
         import sage.all as sage
         return self.real_part()._sage_() + sage.I * self.imaginary_part()._sage_()
 
+cdef class Infinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return sympy.oo
+
+    def _sage_(self):
+        import sage.all as sage
+        return sage.oo
+
+cdef class NegativeInfinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return -sympy.oo
+
+    def _sage_(self):
+        import sage.all as sage
+        return -sage.oo
+
+cdef class ComplexInfinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return sympy.zoo
+
+    def _sage_(self):
+        import sage.all as sage
+        return sage.unsigned_infinity
+
+cdef class NaN(Number):
```

Comment:
I don't think that `return nan` would work as we're defining it as:
`nan = c2py(symengine.Nan)`
which I think would lead to an infinite recursion, similar to using:
```
cdef Number X = _sympify(0)
        return c2py(<RCP[const symengine.Basic]>(symengine.make_rcp_Infty(symengine.rcp_static_cast_Number(X.thisptr))))
```
for the other classes. What else can be done here @isuruf ?